pkg/openthread/contrib/platform_alarm.c

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (C) 2016 José Ignacio Alamos
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @{
+ *
+ * @file
+ * @author  José Ignacio Alamos <jialamos@uc.cl>
+ */
+
+#include <stdint.h>
+
+#include "msg.h"
+#include "openthread/platform/alarm.h"
+#include "ot.h"
+#include "thread.h"
+#include "xtimer.h"
+
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+
+static xtimer_t ot_timer;
+static msg_t ot_alarm_msg;
+
+/* OpenThread will call this for starting an aDt millisecs alarm when current time is aT0 millisecs */
+void otPlatAlarmStartAt(otInstance *aInstance, uint32_t aT0, uint32_t aDt)
+{
+    DEBUG("openthread: otPlatAlarmStartAt: aT0: %i, aDT: %i\n", (int) aT0, (int) aDt);
+    ot_alarm_msg.type = OPENTHREAD_XTIMER_MSG_TYPE_EVENT;
+
+    int dt = aDt * 1000U;
+    if (dt == 0) {
```

Comment:
you can check for ```(aDt == 0)``` here and move the calculation into the xtimer_set_message call below.